File: 2015/AI/AIClient_Python/src/event/EventFactory.py
'''
Created on Dec 16, 2014

@author: scarriere
'''
from event.JoinGameEvent import JoinGameEvent
from aiclient.Singleton import Singleton
from event.QueueController import QueueController
from event.StartGameEvent import StartGameEvent
from event.SendGameInfosEvent import SendGameInfosEvent
from event.UpdateCharacterEvent import UpdateCharacterEvent
from event.MineHitEvent import MineHitEvent
from event.MissileHitEvent import MissileHitEvent
from event.UpdateMissileEvent import UpdateMissileEvent
from event.UpdateBoxEvent import UpdateBoxEvent
from event.EndGameEvent import EndGameEvent

class EventFactory(object):
    @staticmethod
    def generateEvent(message):
        messageParts = message.split(":", 1)
        event = EventFactory.createEvent(messageParts[0])
        if event != None:
            event.fillArguments(messageParts[1])
            queueController = Singleton(QueueController)
            queueController.addIngoingEvent(event)
    
    @staticmethod
    def createEvent(eventType):
        if eventType == "JoinGame":
            return JoinGameEvent()
        elif eventType == "StartGame":
            return StartGameEvent()
        elif eventType == "SendGameInfos":
            return SendGameInfosEvent()
        elif eventType == "UpdateCharacter":
            return UpdateCharacterEvent()
        elif eventType == "MineHit":
            return MineHitEvent()
        elif eventType == "MissileHit":
            return MissileHitEvent()
        elif eventType == "UpdateMissile":
            return UpdateMissileEvent()
        elif eventType == "UpdateBox":
            return UpdateBoxEvent()
        elif eventType == "EndGame":
            return EndGameEvent()
        return None

File: 2015/AI/AIClient_JAVA/src/event/EventFactory.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** EventFactory.java
 ** Factory that generate event
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

public class EventFactory {

	public static void generateEvent(String message) {
		String[] messageParts = message.split(":", 2);
		IngoingEvent event = createEvent(messageParts[0]);
		if (event != null) {
			event.fillArguments(messageParts[1]);
			QueueController.getInstance().addIngoingEvent(event);
		} else {
			//System.out.println("Factory Error: Message invalid");
		}
	}

	private static IngoingEvent createEvent(String type) {
		if (type.equals("JoinGame")) {
			return new JoinGameEvent();
		} else if (type.equals("StartGame")) {
			return new StartGameEvent();
		} else if (type.equals("SendGameInfos")) {
			return new SendGameInfosEvent();
		} else if (type.equals("UpdateCharacter")) {
			return new UpdateCharacterEvent();
		} else if (type.equals("MineHit")) {
			return new MineHitEvent();
		} else if (type.equals("MissileHit")) {
			return new MissileHitEvent();
		} else if (type.equals("UpdateMissile")) {
			return new UpdateMissileEvent();
		} else if (type.equals("UpdateBox")) {
			return new UpdateBoxEvent();
		} else if (type.equals("EndGame")) {
			return new EndGameEvent();
		}
		return null;
	}
}